### PR TITLE
Implement org deletion job endpoint

### PR DESCRIPTION
**Pull request by acosta11**
## Is there a related GitHub Issue?
#2605 

## What is this change about?
Return status of org deletion based on underlying CFOrg resource

## Does this PR introduce a breaking change?
No

## Acceptance Steps
See issue.

## Tag your pair, your PM, and/or team
paired w/ @davewalter

### CODE REVIEW

**Comment by acosta11** [comment]
Messed up my git author config;  had to reset-author and force push to fix the CLA issue.

**Review by akrishna90 — COMMENTED** [review]
LGTM. I am skeptical about setting 120s as timeout. Is that enough? Does this need to be configurable? I guess for now it should be fine until we get some feedback or hear otherwise.

File: api/handlers/job_test.go [hunk 1]
```diff
@@ -81,5 +90,96 @@ var _ = Describe("Job", func() {
...
+			When("org delete is in progress", func() {
+				BeforeEach(func() {
+					orgRepo.GetOrgReturns(repositories.OrgRecord{
+						GUID:      "cf-org-guid",
+						DeletedAt: time.Now().Format(time.RFC3339Nano),
+					}, nil)
+				})
+
+				It("returns a processing status", func() {
+					guid := operation + "~" + resourceGUID
```
Comment (danail-branekov) [inline api/handlers/job_test.go:114]: All that setup could be moved into `BeforeEach` - to initialise the request URL - and `JustBeforeEach` - to execute the request. Thus the setup repetition in `It`s would be removed and the test would be aligned with idiomatic Ginkgo style

**Comment by acosta11** [comment]
> LGTM. I am skeptical about setting 120s as timeout. Is that enough? Does this need to be configurable? I guess for now it should be fine until we get some feedback or hear otherwise.

Yeah Dave and I had the same thought about circling back if we get feedback but think it's a reasonable starting point given we already have a 1 minute timeout on app deletion.

**Review by akrishna90 — APPROVED** [review]

**Pull request merged**